cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -130,14 +136,49 @@ impl ToString for LockedManifest {
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
 pub struct LockedManifestCatalog {
     #[serde(rename = "lockfile-version")]
-    version: Version<1>,
+    pub version: Version<1>,
     /// original manifest that was locked
-    manifest: TypedManifestCatalog,
-    /// locked groups
-    groups: Vec<LockedGroup>,
+    pub manifest: TypedManifestCatalog,
+    /// locked pacakges
+    pub packages: Vec<LockedPackageCatalog>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct LockedPackageCatalog {
+    // region: original fields from the service
+    // These fields are copied from the generated struct.
+    pub attr_path: String,
+    pub broken: bool,
+    pub derivation: String,
+    pub description: String,
+    pub license: String,
+    pub locked_url: String,
+    pub name: String,
+    pub pname: String,
+    pub rev: String,
+    pub rev_count: i64,
+    #[cfg_attr(test, proptest(strategy = "crate::utils::proptest_chrono_strategy()"))]
+    pub rev_date: chrono::DateTime<chrono::offset::Utc>,
+    #[cfg_attr(test, proptest(strategy = "crate::utils::proptest_chrono_strategy()"))]
+    pub scrape_date: chrono::DateTime<chrono::offset::Utc>,
+    pub stabilities: Vec<String>,
+    pub unfree: bool,
+    pub version: String,
+    // endregion
+
+    // region: converted fields
+    pub outputs: BTreeMap<String, String>,
+    pub outputs_to_install: Vec<String>,
+    // endregion
+
+    // region: added fields
+    pub system: System,
+    // endregion
 }
```

Comment:
Currently this is basically a copy of `PackageResolutionInfo` plus `system`
I want to avoid defining the lockfile based on plain response types, especially without any stable shim around them.

Reply:
I dont think it should be in the shim, since we should not define the lockfile based on the response, rather map the response to the lockfile type since we control (and guarantee stability of) the lockfile type.

Metadata from the descriptor (such as e.g. group name, priority, etc) can be recovered by looking up the package by id.

`system` is the only attribute that can not be derived from the rest of the package info, but we need the system to filter out what to build via `buildenv`.